### PR TITLE
Used a cheaper type check in toMap function, to improve performance

### PR DESCRIPTION
**Pull request by mlewand**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Improved `toMap` method performance. This results with improved editor data processing speed. Closes ckeditor/ckeditor5#5854.

---

### Additional information

lodash' `isPlainObject` is a pretty performance expensive function, so should be avoided in places that are called frequently. This PR reverses the condition so a cheaper and more common condition is used first.

Alternatively we could create our own, more optimized `isPlainObject` implementation, but since our code uses iterables for the most part here, it's enough to just reverse the condition check.

You can use manual tests added in ckeditor/ckeditor5#5880 to have use performance tests.

There's a sub pr for this: ckeditor/ckeditor5-engine#1822.

#### tl;dr

| test case | total time (ms/%) | PR total time (ms/%) | total click handler time | PR total click handler time |
| --- | --- | --- | --- | --- |
| [rich editor setData](http://localhost:8125/ckeditor5/tests/manual/performance/richeditor.html) small content | 7.6ms / 5.7% | 0.5ms / 0.4% | 131.92ms | 129.39ms (Reduced by ~1.92%) |
| rich editor setData medium content | 805.4ms / 16.2% | 21.4ms / 0.5% | 4.99s | 4.02s (reduced by ~19.4%) |
| [performance/setdata](http://localhost:8125/ckeditor5/tests/manual/performance/setdata.html) full websites (styled) | 632.2ms / 13.7% | 20.7ms / 0.5% | 4.23s | 3.55s (reduced by ~%16) |

#### RichEditor (short semantic setData)

Before:

After:

#### RichEditor (medium semantic setData)

Before:

After:

#### [setData](http://localhost:8125/ckeditor5/tests/manual/performance/setdata.html) (full websites)

Before:

After:

### CODE REVIEW

**Comment by oleq** [comment]
Guys, make sure these PRs are backed by noticeable performance gain confirmed in Chrome Dev Tools. For instance, do some [heavy editor startup](https://github.com/ckeditor/ckeditor5-utils/pull/322#issuecomment-581962587) check before and after.
